library/spdm_requester_lib/libspdm_req_send_receive.c

```diff
@@ -734,6 +734,27 @@ libspdm_return_t libspdm_receive_spdm_response(libspdm_context_t *spdm_context,
 
     if (spdm_response->request_response_code == SPDM_ERROR
         && spdm_response->param1 == SPDM_ERROR_CODE_LARGE_RESPONSE) {
+        spdm_message_header_t *request_message;
+
+        request_message = (spdm_message_header_t *)spdm_context->last_spdm_request;
+
+        switch (request_message->request_response_code)
+        {
+        case SPDM_GET_VERSION:
+        case SPDM_GET_CAPABILITIES:
+        case SPDM_PSK_FINISH:
+        case SPDM_HEARTBEAT:
+        case SPDM_KEY_UPDATE:
+        case SPDM_END_SESSION:
+        case SPDM_SET_CERTIFICATE:
```

Comment:
SPDM spec only ensures `SPDM_GET_VERSION` and `SPDM_GET_CAPABILITIES` cannot be chunked.

Why we put other here? Those messages may change in the future.

Reply:
If they change in the future then we can put version logic in there. But for now they cannot be chunked.

Reply:
I don't think it is useful to add such check.
1) The list is unnecessary, because the list not defined in the spec.
2) The list is insufficient, because any command/response, whose length is smaller than `DataTransferSize`, should be forbidden and add to the list.

Adding more here just creates confusing to me.

I think we should choose one of below:
1) Just check SPDM_GET_VERSION and SPDM_GET_CAPABILITIES, which is defined in SPDM code.
2) Add a generic check for the response length and `DataTransferSize`.

Reply:
>The list is unnecessary, because the list not defined in the spec.

The specification says
> For SPDM messages that are less than or equal to both the DataTransferSize of the receiving SPDM endpoint and the transmit buffer size of the sending SPDM endpoint, the sending SPDM endpoint shall not utilize this transfer mechanism.

Once the specification has been clarified that the transmit buffer size must be larger than `MinDataTransferSize` then the combination of the above clause plus the fact that the specification defines the size of responses (ie, `KEY_UPDATE_ACK` is defined to be four bytes) then the list is defined in the specification.

>The list is insufficient, because any command/response, whose length is smaller than DataTransferSize, should be forbidden and add to the list.

The Responder's transmit buffer size can be smaller than the Requester's `DataTransferSize` in which case the Responder may reply with `LargeResponse` to a message that is less than the Requester's `DataTransferSize`.

Reply:
I will put this logic in a generic function.